File: packages/node-sdk/src/webhooks/event-types.js
'use strict';

const PAYMENT_SUCCEEDED = 'payment.succeeded';
const PAYMENT_SUCCESS_LEGACY = 'payment.success';
const PAYMENT_FAILED = 'payment.failed';
const PAYMENT_CANCELLED = 'payment.cancelled';
const PAYMENT_CANCELED = 'payment.canceled';
const PAYMENT_EXPIRED = 'payment.expired';
const PAYMENT_REFUNDED = 'payment.refunded';
const REFUND_SUCCEEDED = 'refund.succeeded';
const REFUND_SUCCESS_LEGACY = 'refund.success';
const CHARGEBACK_CREATED = 'chargeback.created';
const DISPUTE_CREATED = 'dispute.created';

function defaultSubscriptions() { return [PAYMENT_SUCCEEDED, PAYMENT_FAILED, PAYMENT_CANCELLED, PAYMENT_EXPIRED, REFUND_SUCCEEDED, PAYMENT_REFUNDED, CHARGEBACK_CREATED, DISPUTE_CREATED]; }
function isSuccessfulPayment(eventType) { return [PAYMENT_SUCCEEDED, PAYMENT_SUCCESS_LEGACY].includes(eventType); }
function isFailedPayment(eventType) { return eventType === PAYMENT_FAILED; }
function isCancelledPayment(eventType) { return [PAYMENT_CANCELLED, PAYMENT_CANCELED, PAYMENT_EXPIRED].includes(eventType); }
function isRefundCompleted(eventType) { return [REFUND_SUCCEEDED, REFUND_SUCCESS_LEGACY, PAYMENT_REFUNDED].includes(eventType); }
function isDisputeOrChargeback(eventType) { return [CHARGEBACK_CREATED, DISPUTE_CREATED].includes(eventType); }

module.exports = { PAYMENT_SUCCEEDED, PAYMENT_SUCCESS_LEGACY, PAYMENT_FAILED, PAYMENT_CANCELLED, PAYMENT_CANCELED, PAYMENT_EXPIRED, PAYMENT_REFUNDED, REFUND_SUCCEEDED, REFUND_SUCCESS_LEGACY, CHARGEBACK_CREATED, DISPUTE_CREATED, defaultSubscriptions, isSuccessfulPayment, isFailedPayment, isCancelledPayment, isRefundCompleted, isDisputeOrChargeback };

File: packages/python-sdk/payxcommerce/webhooks/event_types.py
PAYMENT_SUCCEEDED = "payment.succeeded"
PAYMENT_SUCCESS_LEGACY = "payment.success"
PAYMENT_FAILED = "payment.failed"
PAYMENT_CANCELLED = "payment.cancelled"
PAYMENT_CANCELED = "payment.canceled"
PAYMENT_EXPIRED = "payment.expired"
PAYMENT_REFUNDED = "payment.refunded"
REFUND_SUCCEEDED = "refund.succeeded"
REFUND_SUCCESS_LEGACY = "refund.success"
CHARGEBACK_CREATED = "chargeback.created"
DISPUTE_CREATED = "dispute.created"


def default_subscriptions() -> list[str]:
    return [PAYMENT_SUCCEEDED, PAYMENT_FAILED, PAYMENT_CANCELLED, PAYMENT_EXPIRED, REFUND_SUCCEEDED, PAYMENT_REFUNDED, CHARGEBACK_CREATED, DISPUTE_CREATED]


def is_successful_payment(event_type: str) -> bool:
    return event_type in {PAYMENT_SUCCEEDED, PAYMENT_SUCCESS_LEGACY}


def is_failed_payment(event_type: str) -> bool:
    return event_type == PAYMENT_FAILED


def is_cancelled_payment(event_type: str) -> bool:
    return event_type in {PAYMENT_CANCELLED, PAYMENT_CANCELED, PAYMENT_EXPIRED}


def is_refund_completed(event_type: str) -> bool:
    return event_type in {REFUND_SUCCEEDED, REFUND_SUCCESS_LEGACY, PAYMENT_REFUNDED}


def is_dispute_or_chargeback(event_type: str) -> bool:
    return event_type in {CHARGEBACK_CREATED, DISPUTE_CREATED}
